### PR TITLE
feat: Create EIP 2424

### PR DESCRIPTION
**Pull request by YazzyYaz**
Proposing a new EIP for Code of Conduct in the EIP process to its participants.

This is due to me noticing actions of some individuals in the larger cryptocurrency ecosystem engaging in behavior that would necessitate the existence of such a document.

The Code of Conduct follows Contributor Covenant: https://www.contributor-covenant.org/

which exists in several open-source communities. 

Currently, this Draft EIP will need a new commit to add an appropriate email so victims of harassment within the EIP process have a contact email to send complaints about any fellow member of the EIP process that are violating the Code of Conduct.

Happy to hear any feedback from the EIP community.

It doesn't fully follow the EIP guideline because it is referencing an external document which is licensed under CC-By-4.0 License.

### CODE REVIEW

**Comment by YazzyYaz** [comment]
https://github.com/ethereum/EIPs/issues/2425

**Comment by YazzyYaz** [comment]
For all intents and purposes, this EIP is only limited in scope to the EIP repository and not to any Medium outside of it.

**Comment by YazzyYaz** [comment]
Eth Magicians discussion link: https://ethereum-magicians.org/t/eip-2424-eip-code-of-conduct/3824

**Comment by MicahZoltu** [comment]
@YazzyYaz There should be a single place where discussion occurs, not multiple.  At the moment, the EIP points to a GitHub issue so that is where I'll go but I recommend adding clarity here as to which location is the proper place for discussion.

**Comment by YazzyYaz** [comment]
@MicahZoltu my bad, I can remove the Github discussion link and point to the Ethmagicians if needed. Let me update the markdown.

**Comment by YazzyYaz** [comment]
@MicahZoltu just updated the discussions-to link. Will close the github issue ticket now.

**Comment by 3esmit** [comment]
Would be interesting to require that prior research related to the EIP must be mentioned to reduce plagiarism. 

An emerging problem with a limited set of tools would render similar solutions, it's natural to have researchers overlapping. However, as many work on Ethereum is being done open source, plagiarism can happen and try to look like was a occasional overlap. 

This behavior is toxic as it prevents openness of researchers, and I've seen many cases of open source projects with private GitHub repositories that are revealed only at product launch to prevent plagiarism. 

To de-incentivize such behavior, I would suggest a rule that enforces EIPs to have links to all past research from the creation of draft (date in the EIP), regardless of plagiarism evidence. 
As is difficult to keep track of every project and research in the field, is up to the author of prior research to communicate via comments while not merged, or via issue/PR after is merged, by providing evidence of early research (such as commits timestamped by GitHub).

The community should consider that EIP submitter are actor coordinators, and might not be the author of the research the EIP is about, and this is the case for many EIPs, which correctly mentions all the prior research.

File: EIPS/eip-2424.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+Community leaders have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, and will communicate reasons for moderation decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when an individual is officially representing the community in public spaces. Examples of representing our community include using an official e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported to the community leaders responsible for enforcement at [INSERT CONTACT METHOD]. All complaints will be reviewed and investigated promptly and fairly.
```
Comment (chaals) [inline EIPS/eip-2424.md]: There should be some named contacts here...

Comment (YazzyYaz) [reply EIPS/eip-2424.md]: Agreed, this is why I wrote at the beginning of the PR the following: 
> Currently, this Draft EIP will need a new commit to add an appropriate email so victims of harassment within the EIP process have a contact email to send complaints about any fellow member of the EIP process that are violating the Code of Conduct.

File: EIPS/eip-2424.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Community Impact: Use of inappropriate language or other behavior deemed unprofessional or unwelcome in the community.
+Consequence: A private, written warning from community leaders, providing clarity around the nature of the violation and an explanation of why the behavior was inappropriate. A public apology may be requested.
+
+2. Warning 
+Community Impact: A violation through a single incident or series of actions.
+Consequence: A warning with consequences for continued behavior. No interaction with the people involved, including unsolicited interaction with those enforcing the Code of Conduct, for a specified period of time. This includes avoiding interactions in community spaces as well as external channels like social media. Violating these terms may lead to a temporary or permanent ban.
+
+3. Temporary Ban 
+Community Impact: A serious violation of community standards, including sustained inappropriate behavior.
```
Comment (chaals) [inline EIPS/eip-2424.md]: sustained? how much? I think "repeated" would be clearer (but I don't think the difference is critical).

File: EIPS/eip-2424.md (hunk 1 above)
Comment (chaals) [reply EIPS/eip-2424.md]: Sorry, I was a bit brief. I meant there should be multiple named contacts so people have some choice about who they are dealing with.

**Comment by paulmillr** [comment]
Cannot support this. Code of conducts are great examples of “forced speech” & leftist propaganda. Just be reasonable & don’t harrass folks.

**Comment by shemnon** [comment]
@paulmillr This code of conduct isn't about "leftist propaganda" like what gendered pronouns we use.  I really haven't seen behavior like that in the community that concerns me.  What this CoC push is for is more about how we treat each other and what behaviors are acceptable, outside of what can be labeled as "leftist."  And I have seen behavior in that area that does concern me.  

The Contributor Covenant Open Source Code of Conduct does contain a laundry list of the types of bad behavior that has been seen in other communities, but the real "teeth" of this CoC is in the standards section, which focuses on how we treat each other rather than the politics that have entered other communities.

**Comment by paulmillr** [comment]
@shemnon I would argue that "CoC" bring more politics to the table than it eliminates.

As a matter of fact, I was banned by Google for the first time in the sass organization. Check out issue 910 at sass/dart-sass repo. I was mad at the fact they forced me to sign CLA [(Don't sign a CLA)](https://drewdevault.com/2018/10/05/Dont-sign-a-CLA.html) **after** the contribution was done. This effectively discards contributors. So, I notified folks the process is bogus in a slightly offensive way, but *never harassed anyone*. Got me to re-read their CoC, pretty sure the was no violation.

From my experience, these sorts of things are much more rare in repos, where there is no CoC.

**Comment by shemnon** [comment]
CLA/Contributor License Agreement is an assignment of IP rights (patent, trademark, copyright), completely orthogonal to a Code of Conduct.  This is not proposing a CLA for ethereum, and it is doubtful that one could be made since Ethereum is not a corporation like Google.

**Pull request closed without merging**